### PR TITLE
Changed default cloudmachine.json location

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
CloudMachine assumed its configuration file is in the application root. This worked when the app was hosted in azure (as the cwd is set to the folder with the binaries), but it did not work when debugging locally when by default the project folder is the cwd. 

This change works for both debugging, hosting, and azd.

### CODE REVIEW

File: sdk/core/Azure.Core.Experimental/src/CloudMachine.cs [hunk 1]
```diff
@@ -145,6 +150,8 @@ public void Save(Stream stream)
         [EditorBrowsable(EditorBrowsableState.Never)]
         public void Save(string filepath)
         {
+            var directory = Path.GetDirectoryName(filepath);
```
Comment (annelo-msft) [inline sdk/core/Azure.Core.Experimental/src/CloudMachine.cs:153]: Nit: don't use `var`

File: sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.net461.cs [hunk 2]
```diff
@@ -3,7 +3,7 @@ namespace Azure
     public partial class CloudMachine
     {
         public CloudMachine(System.IO.Stream configurationContent) { }
-        public CloudMachine(string configurationFile = "cloudmachine.json") { }
+        public CloudMachine(string? configurationFile = null) { }
```
Comment (annelo-msft) [inline sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.net461.cs:6]: Curious: why not 

```
public CloudMachine(string configurationFile = ".azure\\cloudmachine.json")
```
?

Comment (KrzysztofCwalina) [reply sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.net461.cs:6]: I want it to run on linux too and thought this is the cleanest way to do it.

File: sdk/core/Azure.Core.Experimental/src/CloudMachine.cs [hunk 3]
```diff
@@ -56,10 +56,15 @@ public static CloudMachine Create(string subscriptionId, string region)
         /// <summary>
         /// Loads CloudMachine settings from configurationFile
         /// </summary>
-        /// <param name="configurationFile"></param>
+        /// <param name="configurationFile">Default value is .azure\cloudmachine.json</param>
         /// <exception cref="InvalidCloudMachineConfigurationException"></exception>
-        public CloudMachine(string configurationFile = "cloudmachine.json")
+        public CloudMachine(string? configurationFile = default)
         {
+            if (configurationFile == null)
```
Comment (annelo-msft) [inline sdk/core/Azure.Core.Experimental/src/CloudMachine.cs:63]: Use `configurationFile ??= "..."` instead?

**Review by annelo-msft — APPROVED** [review]

**Pull request merged**